### ISSUE TITLE
Cannot set initial quality value to achieve a hot start for a quality analysis of type Trace

### ISSUE THREAD

**Issue by pierreAuck** [OpenWaterAnalytics/EPANET#818]
We are trying to run a trace analysis on a large network and update it regularly as new data arrives, using the EPANET toolkit. In order to do that efficiently, we would like store the values at the end of a run, and initialize the initial trace value at each node using the values stored (i.e. hot-start the model).

We were hoping to use EN_INITQUAL, but it seemed do be ignored when doing a trace analysis (and only then, it seems to work fine on water age and chemical analysis). After looking at the code, we saw that indeed setting the initial quality value in that case is explicitely prevented in the source code:
https://github.com/OpenWaterAnalytics/EPANET/blob/fbd005d6044cbd5dd0efddbc933be61c75a1a76c/src/quality.c#L137-L138

Do you know why this limitation has been added? Is there another way to achieve the desired result? I don't understand why EPANET would allow setting initial quality value for chemical simulation and not for trace, to me both seem very similar.

**Comment by LRossman** [comment]
A possible work around is to set up your model to simulate a conservative (non-reactive) chemical whose concentration is a constant value of 100 at the source node. If that node is a Reservoir node then you simply set its initial concentration to 100. If it is a Junction or Tank node then you add a SETPOINT source to it whose baseline value is 100. The simulated concentration results would be interpreted as percent of water from the source node, not as an actual concentration.

**Comment by pierreAuck** [comment]
Thank you for the fast answer! We were thinking of such a workaround but it is reassuring to hear it from you, we will try that.

I still wonder about the reason for this limitation, we would not want to stumble across an issue that was already dealt with in the past. Is it something that could be removed ideally or is there a good reason for it?

**Comment by samhatchett** [comment]
not sure if this is exactly what's being asked, but it is absolutely possible to set network-wide quality conditions on-the-fly using the API. 

My use case was making a long-running simulation to be resilient to process lifecycle interruptions (powercycle / container restart etc.). So when the process executes it will check for existing state saved somewhere and load it back into the epanet model. Snippet here:

https://github.com/OpenWaterAnalytics/epanet-rtx/blob/dev/src/EpanetModel.cpp#L1087-L1117

**Comment by eldemet** [comment]
Hi @pierreAuck, do you think the issue was resolved, in order to close it? Thanks

**Comment by pierreAuck** [comment]
@eldemet not really, I still don't understand why there is this limitation. Yes there is a workaround, but I would prefer to understand the reason behind this specific exclusion of TRACE when setting initial conditions.

**Comment by LRossman** [comment]
@pierreAuck I believe the reason it was decided to not allow an initial condition for a TRACE analysis to be set is that there is no physical way to know what that condition would be. If one claims that it was based on tracer sampling, then you should be modeling the tracer as a CHEMICAL quality constituent and not a TRACE one.

I should mention that trying to update any water quality simulation by restarting it with initial quality set at values from the previous run will not produce a "correct" result since one would also have to initialize the number and quality of the volume segments in each pipe that the previous run ended with. The Toolkit doesn't provide any mechanism to do this.
